arch/x86_64/src/common/Toolchain.defs

```diff
@@ -106,6 +106,12 @@ ifeq ($(CONFIG_DEBUG_LINK_WHOLE_ARCHIVE),y)
   LDFLAGS += --whole-archive
 endif
 
+ifeq ($(CONFIG_HOST_LINUX),y)
```

Comment:
can you update cmake too

Reply:
do you need change Makefile under arch/sim instead arch/x86_64?

Reply:
> do you need change Makefile under arch/sim instead arch/x86_64?

I didn't see warning, when run "./tools/configure.sh -l sim:ostest".
The build design is arch/x86_64 is based on arch/sim ?

Reply:
> can you update cmake too

Done in new commit.

Reply:
if so, CONFIG_HOST_LINUX should drop since CONFIG_HOST_xxx is only meaningful for sim arch.

Reply:
> if so, CONFIG_HOST_LINUX should drop since CONFIG_HOST_xxx is only meaningful for sim arch.

I changed it as the build bug fix : b5d640acc5b3 ("fix Cygwin/MSYS2  ld: unrecognized option '-z'")

Is it about build system ? Since adding it will also make the patch work.

 -l selects the Linux (l) host environment.
  -m selects the macOS (m) host environment.
  -c selects the Windows host and Cygwin (c) environment.
  -g selects the Windows host and MinGW/MSYS environment.